src/Microsoft.ML/Models/BinaryClassificationEvaluator.cs

```diff
@@ -66,7 +68,11 @@ public BinaryClassificationMetrics Evaluate(PredictionModel model, ILearningPipe
                     throw environment.Except($"Could not find ConfusionMatrix in the results returned in {nameof(BinaryClassificationEvaluator)} Evaluate.");
                 }
 
-                return BinaryClassificationMetrics.FromMetrics(environment, overallMetrics, confusionMatrix);
+                var metric = BinaryClassificationMetrics.FromMetrics(environment, overallMetrics, confusionMatrix);
+
+                Contracts.Assert(metric.Count == 1);
+
+                return metric.First();
```

Comment:
It returns a list because that function is retrieves rows from an idataview that contains metrics, for evaluation of a single model it will contain one row but for cross validation it will contain multiple rows that represent all the folds. This function is for single model evaluation case so I take out the first row and only expect one row. Does that answer your question?

---
In reply to: [191001670](https://github.com/dotnet/machinelearning/pull/212#discussion_r191001670) [](ancestors = 191001670)

Reply:
ok, let me rephrase my question, why you do metric.First() instead of metric[0]? 
btw metric.First() will throw if metric is empty which you wouldn't catch through Contracts.Assert in Release build

---
In reply to: [191019339](https://github.com/dotnet/machinelearning/pull/212#discussion_r191019339) [](ancestors = 191019339,191001670)